region/src/main/java/org/motechproject/nms/region/service/impl/LanguageLocationCodesImportServiceImpl.java

```diff
@@ -0,0 +1,219 @@
+                    "Language location exists, yet its circle does not match the circle supplied in the CSV record");
+            verify(Objects.equals(languageLocation.isDefaultForCircle(), defaultForCircle),
+                    "Language location exists, yet its 'defaultForCircle' property does not match the value supplied in the CSV record");
+            languageLocation.getDistrictSet().addAll(districts);
+        } else {
+            verify(!defaultForCircle || null == circle.getDefaultLanguageLocation(),
+                    "Language location with the code '%s' is marked as default for the circle, yet the default language location for this circle already exists", languageLocationCode);
+            languageLocation = new LanguageLocation();
+            languageLocation.setCode(languageLocationCode);
+            languageLocation.setLanguage(language);
+            languageLocation.setCircle(circle);
+            languageLocation.setDefaultForCircle(defaultForCircle);
+            languageLocation.getDistrictSet().addAll(districts);
+            languageLocationDataService.create(languageLocation);
+        }
+    }
+
+    private void verifyStateInCircle(Circle circle, State state) {
+        verify(circle.getStates().contains(state), "State is not contained in the supplied circle");
+    }
+
+    private void verifyDistrictLanguageLocation(District district) {
+        verify(null == district.getLanguageLocation(), "Language location for the '%s' district already specified", district.getName());
+    }
+
+    private void verify(boolean condition, String message, String... args) {
+        if (!condition) {
+            throw new CsvImportDataException(String.format(message, args));
+        }
+    }
+
+    private Map<String, CellProcessor> getProcessorMapping() {
+        Map<String, CellProcessor> mapping = new HashMap<>();
+        mapping.put(LANGUAGE_LOCATION_CODE, new GetString());
+        mapping.put(LANGUAGE, new GetInstanceByString<Language>() {
+            @Override
+            public Language retrieve(String value) {
+                Language language = languageDataService.findByName(value);
+                if (null == language) {
+                    language = languageDataService.create(new Language(value));
+                }
+                return language;
+            }
+        });
+        mapping.put(CIRCLE, new GetInstanceByString<Circle>() {
+            @Override
+            public Circle retrieve(String value) {
+                Circle circle = circleDataService.findByName(value);
+                verify(null != circle, "Circle does not exists");
+                return circle;
+            }
+        });
+        mapping.put(STATE, new Optional(new GetInstanceByLong<State>() {
+            @Override
+            public State retrieve(Long value) {
+                State state = stateDataService.findByCode(value);
```

Comment:
The sample file they gave me has a hidden column that is blank for state code.  It does have the state name.